test-loop-tests/src/tests/catching_up.rs

```diff
@@ -0,0 +1,515 @@
+        peer_actor.register_override_handler(Box::new(move |request| -> Option<NetworkRequests> {
+            if let NetworkRequests::PartialEncodedChunkMessage { partial_encoded_chunk, .. } =
+                &request
+            {
+                if skip_24 {
+                    if partial_encoded_chunk.header.height_created() == 23
+                        || partial_encoded_chunk.header.height_created() == 24
+                    {
+                        return None;
+                    }
+                }
+            }
+            Some(request)
+        }));
+    }
+
+    let client_actor_handle = &env.node_datas[0].client_sender.actor_handle();
+    runner.run_until(
+        &mut env,
+        |test_loop_data| {
+            let client = &test_loop_data.get(client_actor_handle).client;
+            let head = client.chain.head().unwrap();
+            head.height >= send_tx_height
+        },
+        block_prod_time * send_tx_height as i32,
+    );
+
+    let clients = env
+        .node_datas
+        .iter()
+        .map(|test_data| &env.test_loop.data.get(&test_data.client_sender.actor_handle()).client)
+        .collect_vec();
+
+    let mut tx_count = 0;
+    let anchor_hash = get_anchor_hash(&clients);
+    for (i, sender) in test_accounts.iter().enumerate() {
+        for (j, receiver) in test_accounts.iter().enumerate() {
+            let mut amount = (((i + j + 17) * 701) % 42 + 1) as u128;
+            if non_zero {
+                if i > j {
+                    amount = 2;
+                } else {
+                    amount = 1;
+                }
+            }
+            println!("VALUES {:?} {:?} {:?}", sender.to_string(), receiver.to_string(), amount);
+
+            let tx = SignedTransaction::send_money(
+                (12345 + tx_count) as u64,
+                sender.clone(),
+                receiver.clone(),
+                &create_user_test_signer(&sender),
+                amount,
+                anchor_hash,
+            );
+            for node in &env.node_datas {
+                node.rpc_handler_sender.send(ProcessTxRequest {
+                    transaction: tx.clone(),
+                    is_forwarded: false,
+                    check_only: false,
+                });
+            }
+            tx_count += 1;
+        }
+    }
+    assert_eq!(tx_count, 16 * 16);
+
+    runner.run_for_an_epoch(&mut env);
+    runner.run_for_an_epoch(&mut env);
+
+    let mut amounts = HashMap::new();
+
+    assert_eq!(env.node_datas.len(), 16);
+    let view_client_handlers =
+        env.node_datas.iter().map(|data| data.view_client_sender.actor_handle()).collect_vec();
+
+    let client = &env.test_loop.data.get(client_actor_handle).client;
+    let head = client.chain.head().unwrap();
+    let start_height = head.height;
+    let target_height = epoch_length * 6 + 2;
+    runner.run_until(
+        &mut env,
+        |test_loop_data| {
```

Comment:
Can you add a short comment on what is being asserted here?

Reply:
Done in [`43665f1` (#13419)](https://github.com/near/nearcore/pull/13419/commits/43665f1a769c8b9ed65e5cfa996d069074dd5954)